rust_src/src/multibyte.rs

```diff
@@ -403,6 +613,26 @@ impl LispStringRef {
         unsafe { *self.const_data_ptr().offset(index) }
     }
 
+    /// Returns the length of the character sequence at ´index´.
+    pub fn head_len(self, index: usize) -> usize {
+        multibyte_length_by_head(self.as_slice()[index])
+    }
+
+    /// Checks whether `index` is the beginnign of a character sequence (also
+    /// counts single byte characters).
+    pub fn is_head_at(self, index: usize) -> bool {
+        char_head_p(self.as_slice()[index])
+    }
+
+    /// Gets the codepoint starting at the specified index, and its length.
+    ///
+    /// # Panics
```

Comment:
Is there a way to have a version that would use an Option or a Result?

Reply:
Definitely, you just return None if the index isn't a head or if the slice isn't long enough to hold the expected number of bytes.

Reply:
A checked version has been added as `char_and_len_safe`.